presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowColumnStatsRewriteResultBuilder.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.rewrite;
+
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.statistics.ColumnStatistics;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.spi.statistics.TableStatistics;
+import com.facebook.presto.sql.QueryUtil;
+import com.facebook.presto.sql.tree.DoubleLiteral;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.NullLiteral;
+import com.facebook.presto.sql.tree.Row;
+import com.facebook.presto.sql.tree.SelectItem;
+import com.facebook.presto.sql.tree.StringLiteral;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+class ShowColumnStatsRewriteResultBuilder
+{
+    private ShowColumnStatsRewriteResultBuilder() {}
+
+    static List<Expression> buildStatisticsRows(TableStatistics tableStatistics, Set<String> statisticsNames)
+    {
+        ImmutableList.Builder<Expression> rowsBuilder = ImmutableList.builder();
+
+        // Stats for columns
+        for (Map.Entry<ColumnHandle, ColumnStatistics> columnStats : tableStatistics.getColumnStatistics().entrySet()) {
+            Map<String, Estimate> columnStatisticsValues = columnStats.getValue().getStatistics();
+            rowsBuilder.add(createStatsRow(Optional.of(columnStats.getKey().toString()), statisticsNames, columnStatisticsValues));
+        }
+
+        // Stats for whole table
+        rowsBuilder.add(createStatsRow(Optional.empty(), statisticsNames, tableStatistics.getTableStatistics()));
+
+        return rowsBuilder.build();
+    }
+
+    static SelectItem[] buildSelectItems(List<String> columnNames)
+    {
+        return columnNames.stream().map(QueryUtil::unaliasedName).toArray(SelectItem[]::new);
+    }
+
+    static List<String> buildColumnsNames(Set<String> statisticsNames)
+    {
+        ImmutableList.Builder<String> columnNamesBuilder = ImmutableList.builder();
+        columnNamesBuilder.add("column_name");
+        columnNamesBuilder.addAll(statisticsNames);
+        return columnNamesBuilder.build();
+    }
+
+    private static Row createStatsRow(Optional<String> columnName, Set<String> statisticsNames, Map<String, Estimate> columnStatisticsValues)
```

Comment:
It seem to me that we should use `List` instead of `Set` for statisticsName to signal that we care about order.
Modulo fact that this argument should disapear alltogether ;)